### PR TITLE
Update NCEPLIBS and enable OpenMP for SP

### PR DESCRIPTION
**Pull request by kgerheiser**
OpenMP is important for SP so enable it for all the HPC configs.

NCEPLIBS fixes for GCC 10, cleanup CMakeLists, and remove -xHOST from certain libraries

### CODE REVIEW

File: config/stack_custom.yaml
Comment (aerorahul) [inline config/stack_custom.yaml:125]: reason to keep this `OFF`?

**Review by aerorahul — APPROVED** [review]
Thanks for updating the pointers to the NCEPLIBS.
Just one question re. keeping OPENMP OFF for sp in `stack_custom.yaml`.

Otherwise everything looks good.
Will merge after the actions complete.

File: config/stack_custom.yaml
Comment (kgerheiser) [reply config/stack_custom.yaml:125]: No

Is there a reason in general to disable OpenMP by default? I changed SP because I know that SP in particular makes heavy use of it to calculate transformations

Comment (aerorahul) [reply config/stack_custom.yaml:125]: No. There is no reason to turn it OFF.  It was the default in our NCEPLIBS, and I stuck with that in my iteration.
Just like I was building shared libs, but now we have turned them to static.

**Comment by aerorahul** [comment]
@Hang-Lei-NOAA Why did you merge this? 
There is a discussion going on.
Also, the CI actions have not completed.

**Comment by Hang-Lei-NOAA** [comment]
Okay, I reviewed the code. Agreed with it.
The only option is merge.

**Comment by climbfuji** [comment]
> @Hang-Lei-NOAA Why did you merge this?
> There is a discussion going on.
> Also, the CI actions have not completed.

I don't know what you all discussed yesterday, but my feeling is that in general we shouldn't merge changes such as MPI OFF->ON without running a set of tests (most likely ufs-weather-model regression tests) that we previously agreed on. I see it stayed off, so that's good.

To @Hang-Lei-NOAA's defense, @aerorahul approved the PR. Maybe we shouldn't approve while there are ongoing discussions or while the CI tests are still running?

**Comment by Hang-Lei-NOAA** [comment]
Sorry, i did not noticed this.
Rahual reminder me.
I have talked with Kyle to modify the changes.
He will push a fix on this based on discussion.

Hang

**Comment by kgerheiser** [comment]
MPI or OpenMP? I did turn OpenMP on for SP in several configs.

**Comment by climbfuji** [comment]
> MPI or OpenMP? I did turn OpenMP on for SP in several configs.

I am sorry, OpenMP of course. There are so many issues with linking to the correct thread-safe versions of runtime libraries in complex application, one never knows what happens unless you have a comprehensive set of tests that tell you that before-after is the same. The ufs-weather-model also has a threading test (and soon threading unit tests that run for more than just one config) which will allow us to catch any b4b differences when changing the number of threads (there shouldn't be any).

**Comment by Hang-Lei-NOAA** [comment]
Agree,  Several nceplibs libraries are using openmp.
These have been tested with fv3GFS v16 for implementation.

**Comment by climbfuji** [comment]
> Agree, Several nceplibs libraries are using openmp. These have been tested with fv3GFS v16 for implementation.
> […](#)

Ok, that's fine. Is this the case for SP? If so, no problem. Just throwing out some words of caution here. Once we use hpc-stack with the ufs-weather-model develop branch and other ufs codes, we will need to run a set of predefined regression tests before making any updates to the libraries.

**Comment by kgerheiser** [comment]
I can either turn it back off or leave it on.

I'll make a new PR to enable it for others and we can test it before merging.

**Comment by kgerheiser** [comment]
When I sent SP to NCO for implementation OpenMP is enabled

**Comment by aerorahul** [comment]
> > Agree, Several nceplibs libraries are using openmp. These have been tested with fv3GFS v16 for implementation.
> > […](#)
> 
> Ok, that's fine. Is this the case for SP? If so, no problem. Just throwing out some words of caution here. Once we use hpc-stack with the ufs-weather-model develop branch and other ufs codes, we will need to run a set of predefined regression tests before making any updates to the libraries.

@climbfuji I think as we test these and other settings, things like this will settle down.  We rarely will change configurations radically. And even if we did, with a proper approach, we can have multiple stacks with different versions/combinations for development, regression testing and stable purposes.

As for code-review and CI, I view them as independent and complementary (and that depends on the reviewers); Some reviewers won't review until the CI finishes (which is fair).  I reviewed the code and approved it because the changes are very simple to review (I also expect CI to pass).  But, I would not merge the branch before the CI finishes. Just approval is not a sufficient condition to merge when CI is present.  E.g. last week was w3emc (I think).  You and I both approved the gcc-10 changes, but CI failed due to an `apt-get update` issue.

**Comment by Hang-Lei-NOAA** [comment]
Agree with Rahul. I will be more careful.

**Comment by climbfuji** [comment]
> > > Agree, Several nceplibs libraries are using openmp. These have been tested with fv3GFS v16 for implementation.
> > > […](#)
> > 
> > 
> > Ok, that's fine. Is this the case for SP? If so, no problem. Just throwing out some words of caution here. Once we use hpc-stack with the ufs-weather-model develop branch and other ufs codes, we will need to run a set of predefined regression tests before making any updates to the libraries.
> 
> @climbfuji I think as we test these and other settings, things like this will settle down. We rarely will change configurations radically. And even if we did, with a proper approach, we can have multiple stacks with different versions/combinations for development, regression testing and stable purposes.
> 
> As for code-review and CI, I view them as independent and complementary (and that depends on the reviewers); Some reviewers won't review until the CI finishes (which is fair). I reviewed the code and approved it because the changes are very simple to review (I also expect CI to pass). But, I would not merge the branch before the CI finishes. Just approval is not a sufficient condition to merge when CI is present. E.g. last week was w3emc (I think). You and I both approved the gcc-10 changes, but CI failed due to an `apt-get update` issue.

Ok, as long as we all know what to do and when then that's ok.

**Comment by aerorahul** [comment]
Well, look what we have.
macOS action failed. Who wants to look?
@Hang-Lei-NOAA care to revert that merge?

**Comment by Hang-Lei-NOAA** [comment]
Okay, I will revert it

**Comment by aerorahul** [comment]
Screen Shot 2020-09-02 at 1 12 10 PM

Also fails on Orion.

**Pull request merged**
